src/MUIDataTable.js

```diff
@@ -332,7 +332,7 @@ class MUIDataTable extends React.Component {
     let filterData = [];
     let filterList = [];
 
-    if (this.state.columns.length && isEqual(this.rawColumns(newColumns), this.rawColumns(this.props.columns))) {
+    if (this.state.columns.length) {
```

Comment:
As far as I can tell the isEqual code is not needed. All the current calling contexts pass in `this.props.columns`